feat(dev-server): Update default port to 3020 with conflict handling

k0msenapati opened the pull request:
/claim #36
Closes #36

k0msenapati committed:
feat(dev-server): Update default port to 3020 with conflict handling

seveibar reviewed (changes requested):
pretty good but let's tweak it to simplify the code.

- Only initialize fsKy once, this means it will be null prior to the server starting (if this is messy, do an any case `null as any` in the constructor)
- Determine if the port is free prior to the `createHttpServer` function, it's very confusing how you pass in a port that may not be used
- Use a library or generally simplify the port selection. It's too complex.
- In the test, don't start two dev server instances, just bind an http server to 3020 and start a single dev server and ping it at 3021 or whatever to make sure it found a port ok

seveibar commented:
This approach is better https://github.com/tscircuit/cli/pull/37

k0msenapati committed:
refactor: simplify port handling with get-port

- Use get-port library for automatic port selection
- Initialize fsKy only after server starts with final port
- Add proper cleanup of server resources

k0msenapati committed:
test: simplify port conflict test with blocking server

- Replace multi-server setup with single blocking server
- Add direct HTTP request to verify server is running
